gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbDataTable.java

```diff
@@ -0,0 +1,49 @@
+package org.sakaiproject.gradebookng.tool.component;
+
+import org.apache.wicket.extensions.markup.html.repeater.data.table.AbstractColumn;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
+import org.apache.wicket.markup.ComponentTag;
+import org.apache.wicket.markup.repeater.data.IDataProvider;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.model.IModel;
+
+import java.util.List;
+import java.util.Map;
+
+public class GbDataTable<T, S> extends DataTable<T, S> {
+
+	IModel<Map<String, Object>> gradebookData;
+
+	public GbDataTable(final String id, final List<? extends IColumn<T, S>> columns,
+			final IDataProvider<T> dataProvider, final long rowsPerPage, final IModel<Map<String, Object>> model) {
+		super(id, columns, dataProvider, rowsPerPage);
+		this.gradebookData = model;
+	}
+
+
+	@Override
+	protected Item<IColumn<T, S>> newCellItem(final String id, final int index, final IModel<IColumn<T, S>> model) {
+		return new Item(id, index, model) {
```

Comment:
So this is the method it'd be nice to override inline as per `new DataTable() {@Override protected Item<IColumn> newCellItem(......` if only I could figure out the generics :confounded: